src/components/Modal/node/SendMessageModal.tsx

```diff
@@ -67,6 +75,28 @@ type SendMessageModalProps = {
   tooltip?: JSX.Element | string;
 };
 
+function sortAddresses(
+  peers: GetPeersResponseType | null,
+  me: AddressesType,
+  peerIdToAliasLink: {
+    [peerId: string]: string
+  }
+) : string[] {
+  if(!peers || !me) return [];
+  const connectedPeers = peers.connected;
+  const myAddress = me.hopr as string;
+  const peerIdsWithAliases = Object.keys(peerIdToAliasLink);
+  if(peerIdsWithAliases.length === 0) return [myAddress, ...connectedPeers.map(peer => peer.peerId)];
+  let array = [myAddress];
+  peerIdsWithAliases.forEach(peerId => {
+    if(peerId !== myAddress) array.push(peerId)
+  });
+  connectedPeers.forEach(peer => {
+    if(peer.peerId !== myAddress && !peerIdsWithAliases.includes(peer.peerId) ) array.push(peer.peerId)
```

Comment:
includes on an array can be expensive with larger sizes arrays. consider converting into a set before checking includes

Reply:
Used the peerIdToAliasLink which I already have. Good point